synapse/models/geospace.py

```diff
@@ -67,13 +80,13 @@ def postTypeInit(self):
         self.setNormFunc(int, self._normIntStr)
         self.setNormFunc(str, self._normIntStr)
         self.setNormFunc(float, self._normFloat)
+        self.context = getGeospaceContext()
 
     def _normFloat(self, valu):
         if valu > 90.0 or valu < -90.0:
             raise s_exc.BadTypeValu(valu=valu, name=self.name,
                                     mesg='Latitude may only be -90.0 to 90.0')
-
-        valu = int(valu * Latitude.SCALE) / Latitude.SCALE
+        valu = float(self.context.create_decimal(valu))
```

Comment:
why are we storing a float? Wouldn't it be better to store as int * 10 ^ 8?

Reply:
The system-mode value for a geo:latlong is currently a float, so changing that would be a breaking api change.  This creates a python float whose value has been rounded consistently out to 9 decimals.

Reply:
If anything, the index byte generation math should be done with fixed length decimal context.